File: javascript/rna-transcription/rna-transcription.js
var DnaTranscriber = function() {}

DnaTranscriber.prototype.toRna = function(dna) {

    var rna = dna;
    for (var i = 0; i < dna.length; i++) {
        if (dna[i] != 'G' && dna[i] != 'C' && dna[i] != 'A' && dna[i] != 'T') {
            throw new Error('Invalid input');
        }
    }
    rna = rna.replace(/G/g, "L");
    rna = rna.replace(/C/g, "M");
    rna = rna.replace(/A/g, "N");
    rna = rna.replace(/T/g, "O");
    rna = rna.replace(/L/g, "C");
    rna = rna.replace(/M/g, "G");
    rna = rna.replace(/N/g, "U");
    rna = rna.replace(/O/g, "A");
    return rna;
}

module.exports = DnaTranscriber;

File: python/rna-transcription/rna_transcription.py
import re
def to_rna(dna):
    rna = dna
    for i in range (0,len(dna)):
        if dna[i] != 'G' and dna[i] != 'C' and dna[i] != 'A' and dna[i] != 'T':
            return ''
    rna = re.sub(r'G','L', rna);
    rna = re.sub(r'C','M',rna);
    rna = re.sub(r'A','N',rna);
    rna = re.sub(r'T','O',rna);
    rna = re.sub(r'L','C',rna);
    rna = re.sub(r'M','G',rna);
    rna = re.sub(r'N','U',rna);
    rna = re.sub(r'O','A',rna);
    return rna;
    pass